### ISSUE TITLE
Improve error message when JSON serialisation fails

### ISSUE THREAD

**Issue by chrisburr** [DIRACGrid/WebAppDIRAC#490]
When `json.dumps` detects an serialisable type a reasonable error is raised:

```python
In [1]: import json
   ...: json.dumps(set())
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-9061567d8f37> in <module>
      1 import json
----> 2 json.dumps(set())

/opt/dirac/versions/v7.2.10-1624978207/Linux-x86_64/lib/python3.9/json/__init__.py in dumps(obj, skipkeys, ensure_ascii, check_circular, allow_nan, cls, indent, separators, default, sort_keys, **kw)
    229         cls is None and indent is None and separators is None and
    230         default is None and not sort_keys and not kw):
--> 231         return _default_encoder.encode(obj)
    232     if cls is None:
    233         cls = JSONEncoder

/opt/dirac/versions/v7.2.10-1624978207/Linux-x86_64/lib/python3.9/json/encoder.py in encode(self, o)
    197         # exceptions aren't as detailed.  The list call should be roughly
    198         # equivalent to the PySequence_Fast that ''.join() would do.
--> 199         chunks = self.iterencode(o, _one_shot=True)
    200         if not isinstance(chunks, (list, tuple)):
    201             chunks = list(chunks)

/opt/dirac/versions/v7.2.10-1624978207/Linux-x86_64/lib/python3.9/json/encoder.py in iterencode(self, o, _one_shot)
    255                 self.key_separator, self.item_separator, self.sort_keys,
    256                 self.skipkeys, _one_shot)
--> 257         return _iterencode(o, 0)
    258
    259 def _make_iterencode(markers, _default, _encoder, _indent, _floatstr,

/opt/dirac/versions/v7.2.10-1624978207/Linux-x86_64/lib/python3.9/json/encoder.py in default(self, o)
    177
    178         """
--> 179         raise TypeError(f'Object of type {o.__class__.__name__} '
    180                         f'is not JSON serializable')
    181

TypeError: Object of type set is not JSON serializable
```

For some reason `WebHandler.encodeDatetime` prevents this from happening:

```python
In [2]: import datetime
   ...: import json
   ...: from WebAppDIRAC.Lib.WebHandler import encodeDatetime
   ...: json.dumps(set(), default=encodeDatetime)
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-2-8321cb542103> in <module>
      2 import json
      3 from WebAppDIRAC.Lib.WebHandler import encodeDatetime
----> 4 json.dumps(set(), default=encodeDatetime)

/opt/dirac/versions/v7.2.10-1624978207/Linux-x86_64/lib/python3.9/json/__init__.py in dumps(obj, skipkeys, ensure_ascii, check_circular, allow_nan, cls, indent, separators, default, sort_keys, **kw)
    232     if cls is None:
    233         cls = JSONEncoder
--> 234     return cls(
    235         skipkeys=skipkeys, ensure_ascii=ensure_ascii,
    236         check_circular=check_circular, allow_nan=allow_nan, indent=indent,

/opt/dirac/versions/v7.2.10-1624978207/Linux-x86_64/lib/python3.9/json/encoder.py in encode(self, o)
    197         # exceptions aren't as detailed.  The list call should be roughly
    198         # equivalent to the PySequence_Fast that ''.join() would do.
--> 199         chunks = self.iterencode(o, _one_shot=True)
    200         if not isinstance(chunks, (list, tuple)):
    201             chunks = list(chunks)

/opt/dirac/versions/v7.2.10-1624978207/Linux-x86_64/lib/python3.9/json/encoder.py in iterencode(self, o, _one_shot)
    255                 self.key_separator, self.item_separator, self.sort_keys,
    256                 self.skipkeys, _one_shot)
--> 257         return _iterencode(o, 0)
    258
    259 def _make_iterencode(markers, _default, _encoder, _indent, _floatstr,

ValueError: Circular reference detected
```

**Comment by chrisburr** [comment]
@TaykYoku, was this fixed by #492?

**Comment by TaykYoku** [comment]
yes, must be fixed there
